rotkehlchen/db/drivers/server.py

```diff
@@ -144,57 +184,64 @@ def _handle_query(
         while True:
+            return_data = db_messages_pb2.ZMQReturnData()  # type: ignore[attr-defined]  # pylint: disable=no-member
             try:
-                call_data: ZMQCallData = self.zmq_connection.recv_pyobj()
-                if call_data.method.value == DbMethod.INITIALIZE.value:
-                    self._handle_connection(call_data)
-                    self.zmq_connection.send_pyobj(None)
+                call_data_serialized = self.zmq_connection.recv()
+                call_data = db_messages_pb2.ZMQCallData()  # type: ignore[attr-defined]  # pylint: disable=no-member
+                call_data.ParseFromString(call_data_serialized)
+                args = tuple(from_typed_data(data) for data in call_data.args)
+                kwargs = {key: from_typed_data(data) for key, data in call_data.kwargs.items()}
+                if call_data.method == DBMethod.INITIALIZE:
+                    self._handle_connection(call_data, args, kwargs)
+                    self.zmq_connection.send(b'')
                     continue
 
                 try:
                     db_connection = self.db_connections[call_data.db_path]
-                except KeyError:
-                    self.zmq_connection.send_pyobj(ZMQReturnData(
-                        result=None,
-                        error=KeyError(f'No DB connection for {call_data.connection_type} found'),
-                    ).serialize())
-                    continue
-
-                if call_data.method.value == DbMethod.OPEN_CURSOR.value:
-                    if self.db_cursors[call_data.db_path].get(call_data.cursor_name or 'default') is not None:  # noqa: E501
-                        raise KeyError(f'Cursor for {call_data.db_path} already exists')
-                    self.db_cursors[call_data.db_path][call_data.cursor_name or 'default'] = db_connection.cursor()  # noqa: E501
-                    self.zmq_connection.send_pyobj(None)
+                except KeyError as e:
+                    raise KeyError(f'No DB connection for {DBConnectionType(call_data.connection_type)} found') from e  # noqa: E501
+
+                cursor_name = call_data.cursor_name or 'default'
+                if call_data.method == DBMethod.OPEN_CURSOR:
+                    if self.db_cursors[call_data.db_path].get(cursor_name) is not None:
+                        raise KeyError(f'Cursor "{cursor_name}" for {call_data.db_path} already exists')  # noqa: E501
+                    self.db_cursors[call_data.db_path][cursor_name] = db_connection.cursor()
+                    self.zmq_connection.send(b'')
                     continue
-                if call_data.method.value == DbMethod.CLOSE_CURSOR.value:
-                    del self.db_cursors[call_data.db_path][call_data.cursor_name or 'default']
-                    self.zmq_connection.send_pyobj(None)
+                if call_data.method == DBMethod.CLOSE_CURSOR:
+                    del self.db_cursors[call_data.db_path][cursor_name]
```

Comment:
this can raise keyError if the cursor is not created. I would use `.pop` with a default here and avoid that error